hawc/apps/animalv2/models.py

```diff
@@ -0,0 +1,576 @@
+
+    # TODO
+    TEXT_CLEANUP_FIELDS = ()
+
+    treatment = models.ForeignKey(
+        Treatment, on_delete=models.CASCADE, related_name="v2_dose_groups"
+    )
+    dose = models.FloatField(validators=[MinValueValidator(0)])
+    dose_units = models.ForeignKey(
+        "assessment.DoseUnits", on_delete=models.CASCADE, related_name="v2_dose_groups"
+    )
+    created = models.DateTimeField(auto_now_add=True)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    def get_assessment(self):
+        return self.treatment.get_assessment()
+
+    def get_study(self):
+        return self.treatment.get_study()
+
+    # TODO - flat_complete_header_row / flat_complete_data_row
+
+
+# can't just name it Endpoint; that clashes around baseendpoint_ptr
+class AnimalEndpoint(BaseEndpoint):
+    objects = managers.AnimalEndpointManager()
+
+    # TODO
+    TEXT_CLEANUP_FIELDS = ()
+
+    experiment = models.ForeignKey(
+        Experiment, on_delete=models.CASCADE, related_name="v2_endpoints"
+    )
+    name_term = models.ForeignKey(
+        Term,
+        related_name="v2_endpoint_name_terms",
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+    )
+    system = models.CharField(max_length=128, blank=True, help_text="Relevant biological system")
+    system_term = models.ForeignKey(
+        Term,
+        related_name="v2_endpoint_system_terms",
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+    )
+    organ = models.CharField(
+        max_length=128,
+        blank=True,
+        verbose_name="Organ (and tissue)",
+        help_text="Relevant organ or tissue",
+    )
+    organ_term = models.ForeignKey(
+        Term,
+        related_name="v2_endpoint_organ_terms",
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+    )
+    effect = models.CharField(
+        max_length=128, blank=True, help_text="Effect, using common-vocabulary"
+    )
+    effect_term = models.ForeignKey(
+        Term,
+        related_name="v2_endpoint_effect_terms",
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+    )
+    effect_subtype = models.CharField(
+        max_length=128, blank=True, help_text="Effect subtype, using common-vocabulary"
+    )
+    effect_subtype_term = models.ForeignKey(
+        Term,
+        related_name="v2_endpoint_effect_subtype_terms",
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+    )
+    # these next 4 are so-called flexible dropdowns. e.g. anatomical is
+    # "left/right/anterior/Posterior/Cranial/Caudal/Lateral/medial" but we want to let
+    # ppl type something else. do we want to store these (and add custom ones) in a table?
+    # Or just build the UI to show options from a list but also let user type custom ones
+    # (in that case just store it as a string)? leave as string for now.
+    effect_modifier_timing = models.CharField(max_length=128, blank=True)
```

Comment:
I'm not sure what to do with these yet... @swatford fyi - I think we need to get more information on how these are supposed to be used... but thanks for taking a first pass...